torch/utils/data/datapipes/utils/common.py

```diff
@@ -18,6 +19,62 @@
 ]
 
 
+def validate_input_col(fn: Callable, input_col: Optional[Union[int, tuple, list]]):
+    """
+    Checks that function used in a map style datapipe works with the input column
+
+    Args:
+        fn: The function to check.
+        input_col: The input column to check.
+    Returns:
+        None.
+    Raises:
+        TypeError: If the function is not compatible with the input column.
+    """
+    sig = inspect.signature(fn)
+    if isinstance(input_col, (list, tuple)):
+        input_col_size = len(input_col)
+    else:
+        input_col_size = 1
+
+    var_pos_args = [p for p in sig.parameters.values() if
+                          p.kind == inspect.Parameter.VAR_POSITIONAL]
+
+    fn_name = "(unknown)"
+    if hasattr(fn, "__name__"):
+        fn_name = fn.__name__
+    elif inspect.isclass(type(fn)):
+        fn_name = type(fn).__name__
```

Comment:
IMO, this is not going to work. `inspect.isclass(type(fn))` should always return True.

Let's use `str(fn)` if not `hasattr(fn, "__name__")`.